src/backend/core/resource_server/authentication.py

```diff
@@ -0,0 +1,54 @@
+"""Resource Server Authentication"""
+
+import base64
+import binascii
+import logging
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+from mozilla_django_oidc.contrib.drf import OIDCAuthentication
+
+from .backend import ResourceServerBackend
+from .clients import AuthorizationServerClient
+
+logger = logging.getLogger(__name__)
+
+
+class ResourceServerAuthentication(OIDCAuthentication):
+    """Authenticate resource owners using the token received from the authorization server."""
+
+    def __init__(self):
+        super().__init__()
+
+        try:
+            authorization_server = AuthorizationServerClient(
+                url=settings.OIDC_OP_URL,
+                verify_ssl=settings.OIDC_VERIFY_SSL,
+                timeout=settings.OIDC_TIMEOUT,
+                proxy=settings.OIDC_PROXY,
+                endpoint_jwks="/jwks",
+                endpoint_introspection="/checktoken",
+            )
+
+            self.backend = ResourceServerBackend(authorization_server)
+
+        except ImproperlyConfigured:
+            logger.debug("Resource server authentication is not enabled", exc_info=True)
+            self.backend = None
```

Comment:
How about `authorization_server_client`? Best of both worlds 😉

Reply:
Perfect, thus the instance is coherent with the class naming.

Reply:
"@sampaccoud I intentionally left it this way because I want to disable resource server authentication if any of these checks fail. By including the exception info, the developer can trace the origin of the issue. While I'm aware that the typical Python practice is to catch exceptions in a single line, I believe that approach would be overkill in this specific case. It might suggest a design flaw, but this approach feels more appropriate for the situation.

Reply:
I've refactored these lines. Catching the same error originating from `AuthorizationServerClient` or `ResourceServerBackend` in two separates try catch would be overkill. The error logged is explicit enough for you to get its origin.